apps/prairielearn/src/ee/lib/ai-grading/ai-grading-stats.ts

```diff
@@ -0,0 +1,129 @@
+import assert from 'node:assert';
+
+import { z } from 'zod';
+
+import { loadSqlEquiv, queryOptionalRow, queryRows } from '@prairielearn/postgres';
+import { DateFromISOString } from '@prairielearn/zod';
+
+import {
+  type AssessmentQuestion,
+  IdSchema,
+  type RubricItem,
+  RubricItemSchema,
+} from '../../../lib/db-types.js';
+
+const sql = loadSqlEquiv(import.meta.url);
+const GradingJobInfoSchema = z.object({
+  grading_job_id: IdSchema,
+  graded_at: DateFromISOString.nullable(),
+  grading_method: z.enum(['Manual', 'AI']),
+  manual_points: z.number().nullable(),
+  manual_rubric_grading_id: IdSchema.nullable(),
+  grader_name: z.string(),
+  rubric_items: z.array(RubricItemSchema),
+});
+type GradingJobInfo = z.infer<typeof GradingJobInfoSchema>;
+
+export interface AIGradingStats {
+  last_human_grader: string | null;
+  ai_grading_status: 'Graded' | 'Latest' | 'Outdated' | 'None';
+  point_difference: number | null;
+  rubric_difference: (RubricItem & { false_positive: boolean })[] | null;
+}
+
+export type WithAIGradingStats<T> = T & AIGradingStats;
+
+/**
+ * Fills in missing columns for manual grading assessment question page.
+ * This includes organizing information about past graders
+ * and calculating point and/or rubric difference between human and AI.
+ */
+export async function fillInstanceQuestionColumns<T extends { id: string }>(
+  instance_questions: T[],
+  assessment_question: AssessmentQuestion,
+): Promise<WithAIGradingStats<T>[]> {
+  const rubric_modify_time = await queryOptionalRow(
+    sql.select_rubric_time,
+    { rubric_id: assessment_question.manual_rubric_id },
+    DateFromISOString,
+  );
+
+  const instance_question_ids = instance_questions.map((iq) => iq.id);
+  const grading_jobs = await queryRows(
+    sql.select_ai_and_human_grading_jobs_and_rubric,
+    { instance_question_ids },
+    GradingJobInfoSchema.extend({ instance_question_id: IdSchema }),
+  );
```

Comment:
Nit, we can inline this now that it's only used once:

```suggestion
  const grading_jobs = await queryRows(
    sql.select_ai_and_human_grading_jobs_and_rubric,
    { instance_question_ids: instance_questions.map((iq) => iq.id) },
    GradingJobInfoSchema.extend({ instance_question_id: IdSchema }),
  );
```